lib/diagnostic.js

```diff
@@ -5,17 +5,23 @@
 // the minimum value in an array
 // this callback will be passed by the tests to the reduce method, e.g.
 // `let minimum = array.reduce(min);`
-const min = function (/* previous, current */) {
-
+const min = function (previous, current, curIndex, a) {
+  let minNum = 0;
+  for (let i = 0; i < a.length; i++) {
+    if (minNum > current) {
+      minNum = current;
+    }
+  }
+  return minNum;
 };
```

Comment:
I think you might have misinterpreted the question. Because there will only be 2 arguments, all we need to do in this function is compare one thing with the other. Check out this solution:
```
const min = function (previous, current) {
  return previous < current ? previous : current;
};
```